[bryanyee33] iP

bryanyee33 opened the pull request:
# FishStock
> [!NOTE]
> [Why not make everything into food?](https://www.youtube.com/watch?v=iXOpxC5kt7Q)

FishStock reminds you of things to do. It's
- The opposite of Stockfish
- Very sm0rt
- Gives you Bitcoins<sup>*</sup>

All you need to do is:
1. Subscribe to [MrBeast](https://www.youtube.com/channel/UCX6OQ3DkcsbYNE6H8uQQuVA)
2. Ask to be featured
3. **Profit**
4. ~~Remember your task after that~~
5. Done :smirk:

*(Also remember to download the app pls)*

Features:
- [x] Allows you to forget your task
- [x] It's Fast
- [x] and it's **FREE!**<sup>*</sup>

If you're a *Jeve* programmer, you can use it to practice Java too. Here's the `main` method:
```
public class FishStock {
    public static void main(String[] args) {
        new FishStock("./data/tasks.txt").run();
    }
}
```

<sub>*May or may not be true. We are not liable for any losses, trauma, or catastrophic disasters occurring.</sub>

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

bryanyee33 committed:
Rename bot and change initial message

bryanyee33 committed:
Add echo functionality

bryanyee33 committed:
Add listing functionality

bryanyee33 committed:
Add Task class and add checked Task feature

bryanyee33 committed:
Remove unnecessary comment in Task.java

bryanyee33 committed:
Switch to all to Todo task, and add all Task subclasses

bryanyee33 committed:
Refactor getting task from index

bryanyee33 committed:
Add all 3 Task types

bryanyee33 committed:
Add automated testing

bryanyee33 committed:
Refactor error checking into subclasses, and add more error message types

bryanyee33 committed:
Modify expected automated testing output

bryanyee33 committed:
Use BobException

bryanyee33 committed:
Modify expected automated testing output

bryanyee33 committed:
Change to ArrayList

bryanyee33 committed:
Add delete functionality

bryanyee33 committed:
Add new test cases

bryanyee33 committed:
Fix spacing

bryanyee33 committed:
Rename Bob to FishStock, and add to package fishstock

bryanyee33 committed:
Switch to addTask method

bryanyee33 committed:
Add saving feature

bryanyee33 committed:
Merge branch 'branch-Level-7'

bryanyee33 committed:
Switch to LocalDateTime

bryanyee33 committed:
Merge branch 'branch-Level-8'

bryanyee33 committed:
Refactor Ui

bryanyee33 committed:
Refactor code to be more OOP

bryanyee33 committed:
Merge branch 'branch-A-MoreOOP'

bryanyee33 committed:
Move all IO to Ui class

bryanyee33 committed:
Change boolean variable name

bryanyee33 committed:
Tidy formatting

bryanyee33 committed:
Tidy formatting

bryanyee33 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

bryanyee33 committed:
Correct main class path

bryanyee33 committed:
Add JUnit Tests

bryanyee33 committed:
Change archiveBaseName to fishstock

bryanyee33 committed:
Add JavaDoc comments

bryanyee33 committed:
Remove unused code

bryanyee33 committed:
Make Ui.run() static

bryanyee33 committed:
Add find functionality

bryanyee33 committed:
Merge branch 'branch-Level-9'

bryanyee33 committed:
Merge branch 'branch-A-CodingStandard'

bryanyee33 committed:
Merge branch 'branch-A-JavaDoc'

bryanyee33 committed:
Add checkstyle gradle and fix coding style

cyqjoseph commented on src/main/java/fishstock/TaskList.java:
```diff
@@ -0,0 +1,116 @@
+        list.add(task);
+        return task;
+    }
+
```
Consider adding JavaDocs for `findTask`!

chin-herng replied:
I see that the JavaDocs has already been added. Perhaps this is because I am 3-weeks late. Regardless, I would like to add-on on why do `findTasks` deserve a JavaDocs. This is due to a relatively complicated implementation of this method about which people who plan to use/invoke the method does not need to know.

cyqjoseph commented on src/main/java/fishstock/Task.java:
```diff
@@ -0,0 +1,52 @@
+/**
+ * Encapsulates a Task object.
+ */
+abstract class Task {
```
Great use of inheritance!

cyqjoseph commented on src/main/java/fishstock/Ui.java:
```diff
@@ -0,0 +1,93 @@
+                    hasStopped = true;
+                    break;
+                case LIST:
+                    result = list.getList();
```
Consider abstracting out this functionality into seperate functions (ie `listTasks / markTask`)!

cyqjoseph commented on src/main/java/fishstock/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+ */
+class Deadline extends Task {
+    protected static final String COMMAND = "deadline";
+    private static final String byKeyword = " /by ";
```
Consider renaming `byKeyword` to `BY_KEYWORD` as it is a constant!

bryanyee33 replied:
Thanks for your feedbacks!

cyqjoseph reviewed (commented):
Overall, the code generaly follows good style practices. Consider implementing the seperate `Commands` as classes to abstract out the different functionality!

bryanyee33 committed:
Fix coding standards

bryanyee33 committed:
Move main loop to FishStock class

bryanyee33 committed:
Add GUI functionality

bryanyee33 committed:
Fix padding

bryanyee33 committed:
Merge branch 'branch-Level-10'

bryanyee33 committed:
Add GUI Title and Icon

bryanyee33 committed:
Use assertions to check program

Assertions are not used.

Assertions can check assumptions about the program state so that the
runtime can verify them.

Let's,
* use assertion to check whether the file path is valid.
* use assertion to check whether correct Commands are used as arguments
  for TaskList::changeMark() and TaskList::addTask().

bryanyee33 committed:
FishStock.java: Refactor getResponse()

getResponse is too long.

It does not follow the SLAP principle.

Let's,
* split the main part of the function into a Command class.
* put response prefixes into constants for readability.

bryanyee33 committed:
Fix style

bryanyee33 committed:
TaskList.java: Split changeMark() into explicit functions.

changeMark() unifies both marking and unmarking.

The extra parameter is unnecessary.

Split into functions markTask() and unmarkTask() without the command
type.

bryanyee33 committed:
Use assertion on task creation

bryanyee33 committed:
Split Task validity checks into a method

bryanyee33 committed:
Add more variables for readability

bryanyee33 committed:
Add keyword field to CommandType enum

bryanyee33 committed:
Add UserInput.java and use split() for inputs

bryanyee33 committed:
Merge pull request #1 from bryanyee33/branch-A-CodeQuality

Refactors code to improve the code quality

bryanyee33 committed:
Merge branch 'master' into branch-A-Assertions

bryanyee33 committed:
Merge pull request #2 from bryanyee33/branch-A-Assertions

Use assertions

bryanyee33 committed:
Fix TaskList::addTask() merge

bryanyee33 committed:
UserInput: Add assert check for null input

bryanyee33 committed:
Parser::parseCommandType(): Use valueOf() instead

bryanyee33 committed:
Add gradle.yml

bryanyee33 committed:
Add workflow_dispatch to gradle.yml

bryanyee33 committed:
Fix StorageTest

bryanyee33 committed:
Add undo functionality

bryanyee33 committed:
Merge branch 'branch-C-Undo'

bryanyee33 committed:
Update StorageTest as Event dates check moved

bryanyee33 committed:
Add TaskFactory.java and unify Task creation

bryanyee33 committed:
Change StorageTest.java to TaskFactoryTest.java

bryanyee33 committed:
Split enum into Command.java

bryanyee33 committed:
Move Task related classes into new package

bryanyee33 committed:
Fix javadoc comments format

bryanyee33 committed:
SLAP harder

bryanyee33 committed:
SLAP harder

bryanyee33 committed:
Display list of available commands on launch

bryanyee33 committed:
Fix styleguide violations

bryanyee33 committed:
Create LICENSE for Stockfish image use

bryanyee33 committed:
Add DialogBox colors and different fonts

bryanyee33 committed:
Merge branch 'branch-A-BetterGui'

bryanyee33 committed:
Change User.png and user DialogBox color

bryanyee33 committed:
UserInputTest.java: Test splitByKeywords()

bryanyee33 committed:
TaskFactoryTest.java: Move to fishstock.task package

bryanyee33 committed:
Fix GUI issues on window resize

bryanyee33 committed:
Add representative screenshot

bryanyee33 committed:
Rearrange displayed commands order

bryanyee33 committed:
Add help command

bryanyee33 committed:
Add user guide

bryanyee33 committed:
Add help command in Command List

bryanyee33 committed:
Fix user guide rendering

bryanyee33 committed:
Fix table in user guide

bryanyee33 committed:
Fix table in user guide again

bryanyee33 committed:
Print only error message when load save file fails

bryanyee33 committed:
Fix dialogContainer size

bryanyee33 committed:
Reduce image sizes

bryanyee33 committed:
Trim whitespace from images

bryanyee33 committed:
Update .gitignore

chin-herng commented on src/main/java/fishstock/task/Event.java:
```diff
@@ -0,0 +1,99 @@
+        this(description, false, from, to);
+    }
+
+    private static void checkIsValid(String[] splitInput) throws TaskException {
```
Would it be better to use some sort of plural form for `splitInput`? I find it quite necessary to emphasise that the parameter is a collection of parts obtained by splitting the input.

chin-herng commented on src/main/java/fishstock/ui/DialogBox.java:
```diff
@@ -0,0 +1,79 @@
+ */
+public class DialogBox extends HBox {
+    private static final CornerRadii CORNER_RADII = new CornerRadii(10);
+    private static final Background USER_DIALOG_BACKGROUND = new Background(
+            new BackgroundFill(Color.CHOCOLATE, CORNER_RADII, null));
+    private static final Background FISHSTOCK_BACKGROUND = new Background(
```
Perhaps it would be better to use a common prefix instead (meaning `BACKGROUND_USER_DIALOG` and `BACKGROUND_FISHSTOCK` rather than the other way around)? This makes the two associated constants appear together when sorted alphabetically. I noticed the same issue in `Logic.java` too.

chin-herng commented on src/main/java/fishstock/ui/MainWindow.java:
```diff
@@ -0,0 +1,60 @@
+ */
+public class MainWindow extends AnchorPane {
+    private static final String WELCOME_TEXT = "Hello, I'm FishStock.\nI might help if I feel like it.\n"
+            + "\nHere's a list of available commands:\n\n" + Command.toList();
```
I like this line break because it does improve readability by breaking before the `+` operator.

chin-herng reviewed (commented):
LGTM. The code is easy to read and seems to adhere to most coding standards to the extent where I need to nit-pick. I am aware that this review came in 3-week late :(
